test repackaged code structure in Azure pipeline

#401 is refactoring code for packaging, code is working from command line and Eclipse IDE

Need to test in Azure pipeline, make sure unit test coverage, imports, etc. still work after refactoring.

failures for dev branch on first pipeline run:

2022-02-03T12:46:44.1721145Z ======================================================================
2022-02-03T12:46:44.1721595Z ERROR: test_a_thermostat_basic_checkout (tests.test_kumocloud_integration.FunctionalIntegrationTest)
2022-02-03T12:46:44.1722127Z Verify thermostat_basic_checkout on target thermostat.
2022-02-03T12:46:44.1723127Z ----------------------------------------------------------------------
2022-02-03T12:46:44.1723475Z Traceback (most recent call last):
2022-02-03T12:46:44.1723871Z   File "/home/vsts/work/1/s/tests/unit_test_common.py", line 201, in test_a_thermostat_basic_checkout
2022-02-03T12:46:44.1724285Z     self.mod.ThermostatClass, self.mod.ThermostatZone
2022-02-03T12:46:44.1724725Z   File "/home/vsts/work/1/s/thermostatsupervisor/thermostat_common.py", line 954, in thermostat_basic_checkout
2022-02-03T12:46:44.1725115Z     ThermostatZone)
2022-02-03T12:46:44.1725496Z   File "/home/vsts/work/1/s/thermostatsupervisor/thermostat_common.py", line 921, in create_thermostat_instance
2022-02-03T12:46:44.1725913Z     Thermostat.print_all_thermostat_metadata(zone)
2022-02-03T12:46:44.1726336Z   File "/home/vsts/work/1/s/thermostatsupervisor/kumocloud.py", line 126, in print_all_thermostat_metadata
2022-02-03T12:46:44.1726740Z     self.get_all_metadata, [zone, debug])
2022-02-03T12:46:44.1727170Z   File "/home/vsts/work/1/s/thermostatsupervisor/thermostat_common.py", line 55, in exec_print_all_thermostat_metadata
2022-02-03T12:46:44.1727564Z     return_data = func(*args)
2022-02-03T12:46:44.1727931Z   File "/home/vsts/work/1/s/thermostatsupervisor/kumocloud.py", line 73, in get_all_metadata
2022-02-03T12:46:44.1728327Z     return self.get_metadata(zone, None, debug)
2022-02-03T12:46:44.1728708Z   File "/home/vsts/work/1/s/thermostatsupervisor/kumocloud.py", line 108, in get_metadata
2022-02-03T12:46:44.1729070Z     self.serial_number = units[zone]
2022-02-03T12:46:44.1729865Z IndexError: list index out of range
2022-02-03T12:46:44.1729995Z 
2022-02-03T12:46:44.1730283Z ======================================================================
2022-02-03T12:46:44.1730704Z ERROR: test_supervise (tests.test_kumocloud_integration.SuperviseIntegrationTest)
2022-02-03T12:46:44.1731091Z Verify supervisor loop on target thermostat.
2022-02-03T12:46:44.1731623Z ----------------------------------------------------------------------
2022-02-03T12:46:44.1731970Z Traceback (most recent call last):
2022-02-03T12:46:44.1732332Z   File "/home/vsts/work/1/s/tests/unit_test_common.py", line 276, in test_supervise
2022-02-03T12:46:44.1732714Z     debug=True, argv_list=self.unit_test_argv)
2022-02-03T12:46:44.1733092Z   File "/home/vsts/work/1/s/thermostatsupervisor/supervise.py", line 210, in exec_supervise
2022-02-03T12:46:44.1733522Z     supervisor(api.user_inputs["thermostat_type"], api.user_inputs["zone"])
2022-02-03T12:46:44.1733951Z   File "/home/vsts/work/1/s/thermostatsupervisor/supervise.py", line 128, in supervisor
2022-02-03T12:46:44.1734327Z     Zone = mod.ThermostatZone(Thermostat)
2022-02-03T12:46:44.1734685Z   File "/home/vsts/work/1/s/thermostatsupervisor/kumocloud.py", line 172, in __init__
2022-02-03T12:46:44.1735040Z     Thermostat_obj.zone_number)
2022-02-03T12:46:44.1735412Z   File "/home/vsts/work/1/s/thermostatsupervisor/kumocloud.py", line 73, in get_all_metadata
2022-02-03T12:46:44.1735809Z     return self.get_metadata(zone, None, debug)
2022-02-03T12:46:44.1736178Z   File "/home/vsts/work/1/s/thermostatsupervisor/kumocloud.py", line 108, in get_metadata
2022-02-03T12:46:44.1736543Z     self.serial_number = units[zone]
2022-02-03T12:46:44.1736830Z IndexError: list index out of range
2022-02-03T12:46:44.1736956Z 
2022-02-03T12:46:44.1737238Z ======================================================================
2022-02-03T12:46:44.1737638Z ERROR: tests.test_supervisor_flask_server (unittest.loader._FailedTest)
2022-02-03T12:46:44.1738452Z ----------------------------------------------------------------------
2022-02-03T12:46:44.1738870Z ImportError: Failed to import test module: tests.test_supervisor_flask_server
2022-02-03T12:46:44.1739228Z Traceback (most recent call last):
2022-02-03T12:46:44.1739632Z   File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/unittest/loader.py", line 436, in _find_test_path
2022-02-03T12:46:44.1740134Z     module = self._get_module_from_name(name)
2022-02-03T12:46:44.1740564Z   File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/unittest/loader.py", line 377, in _get_module_from_name
2022-02-03T12:46:44.1740951Z     __import__(name)
2022-02-03T12:46:44.1741296Z   File "/home/vsts/work/1/s/tests/test_supervisor_flask_server.py", line 25, in <module>
2022-02-03T12:46:44.1741688Z     @unittest.skipIf(not util.is_interactive_environment(),
2022-02-03T12:46:44.1742121Z   File "/home/vsts/work/1/s/thermostatsupervisor/utilities.py", line 306, in is_interactive_environment
2022-02-03T12:46:44.1742548Z     raise OSError(f"unrecognized environment: {parent}")
2022-02-03T12:46:44.1742873Z OSError: unrecognized environment: bash
2022-02-03T12:46:44.1743010Z 
2022-02-03T12:46:44.1743084Z 
2022-02-03T12:46:44.1743366Z ======================================================================
2022-02-03T12:46:44.1743777Z ERROR: test_is_interactive_environment (tests.test_utilities.EnvironmentTests)
2022-02-03T12:46:44.1744149Z Verify is_interactive_environment().
2022-02-03T12:46:44.1744640Z ----------------------------------------------------------------------
2022-02-03T12:46:44.1744984Z Traceback (most recent call last):
2022-02-03T12:46:44.1745365Z   File "/home/vsts/work/1/s/tests/test_utilities.py", line 30, in test_is_interactive_environment
2022-02-03T12:46:44.1745771Z     return_val = util.is_interactive_environment()
2022-02-03T12:46:44.1746174Z   File "/home/vsts/work/1/s/thermostatsupervisor/utilities.py", line 306, in is_interactive_environment
2022-02-03T12:46:44.1746593Z     raise OSError(f"unrecognized environment: {parent}")
2022-02-03T12:46:44.1746924Z OSError: unrecognized environment: bash
2022-02-03T12:46:44.1747059Z 
2022-02-03T12:46:44.1747343Z ======================================================================
2022-02-03T12:46:44.1747763Z FAIL: test_get_meta_data (tests.test_kumocloud_integration.FunctionalIntegrationTest)
2022-02-03T12:46:44.1748110Z Verify get_metadata().
2022-02-03T12:46:44.1748580Z ----------------------------------------------------------------------
2022-02-03T12:46:44.1748918Z Traceback (most recent call last):
2022-02-03T12:46:44.1749281Z   File "/home/vsts/work/1/s/tests/unit_test_common.py", line 260, in test_get_meta_data
2022-02-03T12:46:44.1749621Z     expected_return_type))
2022-02-03T12:46:44.1750250Z AssertionError: False is not true : parameter='address', value=None, metadata is type '<class 'NoneType'>', expected type '<class 'str'>'
2022-02-03T12:46:44.1750530Z 
2022-02-03T12:46:44.1750963Z ----------------------------------------------------------------------
2022-02-03T12:46:44.1751296Z Ran 97 tests in 509.300s
2022-02-03T12:46:44.1751410Z 
2022-02-03T12:46:44.1751661Z FAILED (failures=1, errors=4, skipped=34)

--- a/thermostatsupervisor/utilities.py
+++ b/thermostatsupervisor/utilities.py
@@ -299,6 +299,8 @@ def is_interactive_environment():
     parent = psutil.Process(os.getpid()).parent().name()
     if parent == "cmd.exe":
         return False
+    elif parent == "bash":
+        return False
     elif parent == "eclipse.exe":
         return True
     else:

